uniparc-rest/src/test/java/org/uniprot/api/uniparc/controller/UniParcBestGuessControllerIT.java

```diff
@@ -226,60 +269,121 @@ void bestGuessReturnBadRequestIfFoundDuplicatedEntries() throws Exception {
                         jsonPath(
                                 "$.messages.*",
                                 contains(
-                                        "Invalid request received. More than one Best Guess found {UPI0000183A10:trembl0;UPI0000183A12:trembl2;UPI0000183A13:trembl3}. Review your query and/or contact us.")));
+                                        "Invalid request received. More than one Best Guess found {UPI0000183A12:trembl2;UPI0000183A10:trembl0;UPI0000183A13:trembl3}. Review your query and/or contact us.")));
     }
 
     private void saveEntries() throws IOException, SolrServerException {
         UniParcEntryConverter converter = new UniParcEntryConverter();
 
         // SWISSPROT
+        String uniParcId = "UPI0000183A10";
         List<UniParcCrossReference> crossReferences = new ArrayList<>();
-        crossReferences.add(getXref(UniParcDatabase.SWISSPROT, "swissProt0", 9606, true));
-        crossReferences.add(getXref(UniParcDatabase.TREMBL, "trembl0", 9607, true));
         crossReferences.add(
-                getXref(UniParcDatabase.SWISSPROT_VARSPLIC, "isoformInactive0", 9608, false));
-        crossReferences.add(getXref(UniParcDatabase.EMBL, "EMBL0", 9609, true));
+                UniParcCrossReferenceMocker.createUniParcCrossReference(
+                        UniParcDatabase.SWISSPROT, "swissProt0", 9606, true));
+        crossReferences.add(
+                UniParcCrossReferenceMocker.createUniParcCrossReference(
+                        UniParcDatabase.TREMBL, "trembl0", 9607, true));
+        crossReferences.add(
+                UniParcCrossReferenceMocker.createUniParcCrossReference(
+                        UniParcDatabase.SWISSPROT_VARSPLIC, "isoformInactive0", 9608, false));
+        crossReferences.add(
+                UniParcCrossReferenceMocker.createUniParcCrossReference(
+                        UniParcDatabase.EMBL, "EMBL0", 9609, true));
 
-        UniParcEntry entry = createEntry("UPI0000183A10", 20, crossReferences);
+        UniParcEntry entry = createUniParcEntry(uniParcId, 20, crossReferences);
         UniParcDocument doc = documentConverter.convert(converter.toXml(entry));
         cloudSolrClient.addBean(SolrCollection.uniparc.name(), doc);
-        storeClient.saveEntry(entry);
+        // save uniparc light in store
+        UniParcEntryLight lightEntry =
+                UniParcEntryMocker.createEntryLightWithSequenceLength(
+                        uniParcId, 20, crossReferences.size());
+        uniParcLightStoreClient.saveEntry(lightEntry);
+        // save cross references in store
+        String xrefBatchKey = uniParcId + "_0";
+        uniParcCrossReferenceStoreClient.saveEntry(
```

Comment:
I don't think we can directly use `UniParcCrossReferenceMocker.createUniParcCrossReferencePairs`, as it generates random cross references. For BestGuess, we need the cross references to be created in a specific way to cover various cases. This is important because in `UniParcBestGuessControllerIT.saveEntries`, we are creating cross references in a customized manner to properly test the BestGuess functionality. I believe you had originally added this saveEntries method.